lib/dynflow/dispatcher/client_dispatcher.rb

```diff
@@ -132,19 +132,26 @@ def start_termination(*args)
       end
 
       def dispatch_request(request, client_world_id, request_id)
+        ignore_unknown = false
         executor_id = match request,
                             (on ~Execution do |execution|
                                AnyExecutor
                              end),
                             (on ~Event do |event|
+                               ignore_unknown = event.optional
                                find_executor(event.execution_plan_id)
                              end),
                             (on Ping.(~any, ~any) | Status.(~any, ~any) do |receiver_id, _|
                                receiver_id
                              end)
         envelope = Envelope[request_id, client_world_id, executor_id, request]
         if Dispatcher::UnknownWorld === envelope.receiver_id
-          raise Dynflow::Error, "Could not find an executor for #{envelope}"
+          raise Dynflow::Error, "Could not find an executor for #{envelope}" unless ignore_unknown
+
+          message = "Could not find an executor for optional #{envelope}, discarding."
+          log(Logger::DEBUG, message)
+          respond(envelope, Failed[message])
+          return
```

Comment:
**optional** do we want to ignore all the errors on `optional`? Probably not, because optional is not really optional, but `ignore_non_existent_execution`, right? If it would really be optional, we could just change the `rescue` block to handle both cases :thinking:

Reply:
> Probably not, because optional is not really optional, but ignore_non_existent_execution, right?

Correct :/